### ISSUE TITLE
-Wshift-negative-value in drivers/dma/mv_xor_v2.c

### ISSUE THREAD

**Issue by Nathan-Huckleberry** [ClangBuiltLinux/linux#521]
Undefined behavior caused by shifting negative.
```
drivers/dma/mv_xor_v2.c:264:40: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
        reg &= (~MV_XOR_V2_DMA_IMSG_THRD_MASK << MV_XOR_V2_DMA_IMSG_THRD_SHIFT);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
drivers/dma/mv_xor_v2.c:271:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
        reg &= (~MV_XOR_V2_DMA_IMSG_TIMER_THRD_MASK <<
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
```

**Comment by nickdesaulniers** [comment]
In both cases, the right hand operator of the left shift is zero.  Seems like maybe the defines and shifting operations can be deleted?

**Comment by nickdesaulniers** [comment]
patch: https://groups.google.com/forum/#!topic/clang-built-linux/7oRG4seCcpI

**Comment by nickdesaulniers** [comment]
Please send a v2 w/o parens.

**Comment by nickdesaulniers** [comment]
accepted: http://git.infradead.org/users/vkoul/slave-dma.git/commit/698f7a9be9b2389384c69750d87a0c68202c0621

**Comment by nickdesaulniers** [comment]
fixed: 698f7a9be9b2389384c69750d87a0c68202c0621
